packages/cli/src/commands/generate/auth/templates/azureAd.auth.js.template

```diff
@@ -0,0 +1,28 @@
+// Define what you want `currentUser` to return throughout your app. For example,
+// to return a real user from your database, you could do something like:
+//
+//   export const getCurrentUser = async ({ email }) => {
+//     return await db.user.findOne({ where: { email } })
+//   }
+
+import { AuthenticationError } from '@redwoodjs/api'
+
+export const getCurrentUser = async (decoded, { token, type }) => {
+  const { aud, iss, exp, name, sub, preferred_username: email } = decoded
+
+  // Example: Use e-mail address
+  // const user = await db.user.findOne({ email: { email } })'
+
+  // return user ?? {}
+
+  return { email, name, sub, aud, iss, exp }
+}
+
+// Use this function in your services to check that a user is logged in, and
+// optionally raise an error if they're not.
+
+export const requireAuth = () => {
```

Comment:
It **looks** like AzureAD [supports roles](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#user-and-application-tokens) with a `roles` claim on the token.

If so your template may want to include the "role" version of require auth:

https://github.com/redwoodjs/redwood/blob/main/packages/cli/src/commands/generate/auth/templates/auth.js.template#L104

`parseJWT` should still find those with the `roles` claim at same level as `sub`:

https://github.com/redwoodjs/redwood/blob/main/packages/api/src/parseJWT.ts#L11

```
  return (
    token.decoded?.roles ||
    metadata?.roles ||
    metadata.authorization?.roles ||
    []
  )
```

Reply:
@dthyresson Agreed! I will give this a go along with JWT validation. Thanks for the review and references!

Reply:
Marking as resolved due to outdated.

/cc @dthyresson